internal/datacoord/segment_info.go

```diff
@@ -123,6 +281,8 @@ func (s *SegmentsInfo) DropSegment(segmentID UniqueID) {
 // set the logPath of segement in meta empty, to save space
 // if segment has logPath, make it empty
 func (s *SegmentsInfo) SetSegment(segmentID UniqueID, segment *SegmentInfo) {
```

Comment:
A batch update is protected by using keylock in `UpdateSegmentsInfo` and `UpdateDropChannelSegmentInfo` method of `meta`.
But `SetSegment` is executed in a for-loop one by one.
Therefore, read-only methods can see the half-way states of batch segment info updates.
So we need a `SetSegments` method to avoid it?

Reply:
but is there a special case we need consistent view between different segmets?

Reply:
I'm not sure currently.
Previous single-lock implementation, the batch query operation had this guarantee.
It seems that there's only `compaction`, `dropVChannel`, `updateStartPosition` will change cross segment info.